apps/demo/src/components/scrcpy/state.tsx

```diff
@@ -300,41 +311,71 @@ export class ScrcpyPageState {
                 })
             );
 
-            RECORD_STATE.recorder = new MuxerStream();
+            RECORD_STATE.recorder = new MatroskaMuxingRecorder();
+            RECORD_STATE.videoMetadata = undefined;
+            RECORD_STATE.audioMetadata = undefined;
+
             let lastKeyframe = 0;
-            client.videoStream
-                .pipeThrough(
-                    new InspectStream(
-                        action((packet: ScrcpyVideoStreamPacket) => {
-                            if (packet.type === "configuration") {
-                                const { croppedWidth, croppedHeight } =
-                                    packet.data;
-                                this.log.push(
-                                    `[client] Video size changed: ${croppedWidth}x${croppedHeight}`
-                                );
-
-                                this.configuration = packet;
-                                this.width = croppedWidth;
-                                this.height = croppedHeight;
-                            } else if (packet.keyframe) {
-                                if (lastKeyframe) {
+            client.videoStream.then(({ stream, metadata }) => {
+                runInAction(() => {
+                    RECORD_STATE.videoMetadata = metadata;
+                });
+
+                stream
+                    .pipeThrough(
+                        new InspectStream(
```

Comment:
Prettier adds too much indentation, it's so ugly.

Maybe extract this stream into a variable.